### ISSUE TITLE
QuakeML compatibility

### ISSUE THREAD

**Issue by markhamnr** [crotwell/seisplotjs#13]
It seems to me that seisplot.js's QuakeML parser doesn't support all possible fields.  For example, https://quake.ethz.ch/quakeml/docs/xml?action=AttachFile&do=view&target=QuakeML-BED-1.2.xsd indicates that an <arrival> element may have a <distance> child element, but there doesn't seem to be any way to access that distance with seisplot.js.  Have I misunderstood?  Or is this something that could be added?

**Comment by crotwell** [comment]
Yes, the parser is not complete. I believe I parse all of the fields used by IRIS and USGS fdsn event web services, but not everything.

Probably it would be good to add all possible fields, but unlikely I can do that in the short term. If there are ones that you know are widely used, let me know and I can try to get those in sooner. And of course a pull request is always welcome.

Is there a fdsn event web service that populates these, or is this a quakeml you are generating? Having a way to test would be helpful.

**Comment by markhamnr** [comment]
OK, that makes sense.  As always, the first step was to verify that I wasn't crazy. :)

My interest is mainly in parsing QuakeML from my company's ([ISTI](https://www.isti.com/)) own FDSN service.  I'll check with my bosses about the possibility of submitting a pull request for at least some of the fields in the spec.  I'm cautiously optimistic!

**Comment by crotwell** [comment]
Sorry for long delay on pushing out a release. Finally got some time to play with this again and have a concern/thought. Quakeml does a lot of complex types to allow uncertinties, but the vast majority of the time users want just the value. Seems annoying to have to type `origin.time.value` to get the origin time. I am wondering if it would be a better user experience to make `time` be a getter that returns the `time.value` and allow access to the full TimeQuantity via another name, maybe timeQuantity? Similar for the RealQuanitity used for latitude, longitude and depth. Maybe also time on Pick and mag on Magnitude. This would also make the additions a bit less of a breaking change.

I know it has been a while, but how much would this effect your existing code if I made this change?

**Comment by markhamnr** [comment]
That sounds smart.  Makes sense not to break anything if we can help it.  I really haven't written anything that would relu on the new release yet, so feel free to do whatever makes sense!

**Comment by crotwell** [comment]
prerelease on npmjs

`npm i seisplotjs@3.1.0-alpha`

**Comment by markhamnr** [comment]
How does the release process work?  ISTI is a little wary of pulling in a dependency that's marked alpha.  Is it likely to be promoted to a release or at least a release candidate soon?

**Comment by crotwell** [comment]
I was working on a couple of other items like geojson from usgs, and so was holding off on the next release. If you need it soon I can try to get it done sooner.

**Comment by crotwell** [comment]
I was working on a couple of other items like geojson from usgs, and so was holding off on the next release. If you need it soon I can try to get it done sooner. What is your timeline?

**Comment by crotwell** [comment]
Released... 
https://www.npmjs.com/package/seisplotjs
